immutables/_map.c

```diff
@@ -3414,7 +3414,11 @@ PyTypeObject _Map_Type = {
     .tp_iter = (getiterfunc)map_tp_iter,
     .tp_dealloc = (destructor)map_tp_dealloc,
     .tp_getattro = PyObject_GenericGetAttr,
-    .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,
+    .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC
+    #ifdef Py_TPFLAGS_MAPPING
+        | Py_TPFLAGS_MAPPING
+    #endif
```

Comment:
Note: I am not a C developer. I noticed errors for this name not being available on <3.10, and scrambled this together. Please let me know if this is not how things are done, or if you think the formatting looks off.